ambry-router/src/main/java/com.github.ambry.router/ValidatedByteRange.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+/**
+ * a byte range with defined start and end offsets that are verified to be within the supplied total blob size.
+ */
+class ValidatedByteRange {
```

Comment:
I'm not convinced that we need this class. 

As an alternate, what if you validate `ByteRange` when you got the metadata blob (and throw an `Exception` if validation fails) and then exposed setters in `ByteRange` to set the start/end offsets once they are determined? I'm not sure we need a new class for that purpose.